Have PluginDocument express its style through a stylesheet rather than inline style

smfr opened the pull request:
#### 25fb6b6518a0ba1f00dbdbeb0b630639db1b8aa0
<pre>
Have PluginDocument express its style through a stylesheet rather than inline style
<a href="https://bugs.webkit.org/show_bug.cgi?id=262408">https://bugs.webkit.org/show_bug.cgi?id=262408</a>
rdar://116257438

Reviewed by Tim Horton and Richard Robinson.

In the plug-in document, create a `&lt;head&gt;` element which contains a `&lt;style&gt;` element,
and set the text contents of the `style` element to style which is equivalent to the
inline style attributes that the old code was using.

* Source/WebCore/html/PluginDocument.cpp:
(WebCore::PluginDocumentParser::createStyleElement):
(WebCore::PluginDocumentParser::createDocumentStructure): Deleted.
(WebCore::PluginDocumentParser::appendBytes): Deleted.
(WebCore::PluginDocument::PluginDocument): Deleted.
(WebCore::PluginDocument::createParser): Deleted.
(WebCore::PluginDocument::pluginWidget): Deleted.
(WebCore::PluginDocument::setPluginElement): Deleted.
(WebCore::PluginDocument::detachFromPluginElement): Deleted.
(WebCore::PluginDocument::cancelManualPluginLoad): Deleted.

Canonical link: <a href="https://commits.webkit.org/268684@main">https://commits.webkit.org/268684@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb377b502d44804f69b3d803e704b8a3ba1ff881

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20354 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20782 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21426 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22252 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/18987 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20585 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24039 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20954 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20573 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20460 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17700 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23102 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17626 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18507 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/24793 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18706 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18683 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22729 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19255 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18471 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4894 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/22805 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19080 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/html/PluginDocument.cpp:
```diff
@@ -62,19 +66,44 @@ class PluginDocumentParser final : public RawDataDocumentParser {
+{
+    auto styleElement = HTMLStyleElement::create(document);
+
+    constexpr auto styleSheetContents = R"CONTENTS(
```
what's `CONTENTS` mean here?

aprotyas replied:
It's the arbitrary delimiter for the string literal. We also have `DL`/`PSONRESOURCE`/etc.

aprotyas commented on Source/WebCore/html/PluginDocument.cpp:
```diff
@@ -62,19 +66,44 @@ class PluginDocumentParser final : public RawDataDocumentParser {
+{
+    auto styleElement = HTMLStyleElement::create(document);
+
+    constexpr auto styleSheetContents = R"CONTENTS(
```
SMFR prefix? 😛

rr-codes commented on Source/WebCore/html/PluginDocument.cpp:
```diff
@@ -85,29 +114,10 @@ void PluginDocumentParser::createDocumentStructure()
-    body->setAttributeWithoutSynchronization(marginwidthAttr, "0"_s);
-    body->setAttributeWithoutSynchronization(marginheightAttr, "0"_s);
-#if PLATFORM(IOS_FAMILY)
-    constexpr auto bodyBackgroundColor = SRGBA<uint8_t> { 217, 224, 233 };
```
oh I remember typing this!

cdumez commented on Source/WebCore/html/PluginDocument.cpp:
```diff
@@ -62,19 +66,44 @@ class PluginDocumentParser final : public RawDataDocumentParser {
 
     void appendBytes(DocumentWriter&, const uint8_t*, size_t) final;
     void createDocumentStructure();
+    Ref<HTMLStyleElement> createStyleElement(Document&);
```
Can this be static?

cdumez commented on Source/WebCore/html/PluginDocument.cpp:
```diff
@@ -62,19 +66,44 @@ class PluginDocumentParser final : public RawDataDocumentParser {
+html { width: 100%; height: 100%; }
+body { width: 100%; height: 100%; margin: 0; overflow: hidden; }
+embed { width: 100%; height: 100%; }
+)CONTENTS";
```
Can't we use `_s` suffix to make it an ASCIILiteral? It would avoid the call to `String::fromLatin1()` below.

smfr committed:
Have PluginDocument express its style through a stylesheet rather than inline style
https://bugs.webkit.org/show_bug.cgi?id=262408
rdar://116257438

Reviewed by Tim Horton and Richard Robinson.

In the plug-in document, create a `<head>` element which contains a `<style>` element,
and set the text contents of the `style` element to style which is equivalent to the
inline style attributes that the old code was using.

* Source/WebCore/html/PluginDocument.cpp:
(WebCore::PluginDocumentParser::createStyleElement):
(WebCore::PluginDocumentParser::createDocumentStructure): Deleted.
(WebCore::PluginDocumentParser::appendBytes): Deleted.
(WebCore::PluginDocument::PluginDocument): Deleted.
(WebCore::PluginDocument::createParser): Deleted.
(WebCore::PluginDocument::pluginWidget): Deleted.
(WebCore::PluginDocument::setPluginElement): Deleted.
(WebCore::PluginDocument::detachFromPluginElement): Deleted.
(WebCore::PluginDocument::cancelManualPluginLoad): Deleted.

Canonical link: https://commits.webkit.org/268684@main
